### ISSUE TITLE
Support `ts-toolbelt` types

### ISSUE THREAD

**Issue by eranelbaz** [vega/ts-json-schema-generator#1028]
Hey!
We are using `ts-json-schema-generator` for our api validation - 
but looks like when running `ts-json-schema-generator` with https://github.com/millsp/ts-toolbelt types it generates nothing

Hope you can add support for that!

types such as -
```
 O.Either<
  {
    title: string;
    titleId: string;
  },
  'title' | 'titleId'
>;
```

Thanks

**Comment by arthurfiorette** [comment]
Thanks for the report. Can you provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) that demonstrates the issue?

**Comment by eranelbaz** [comment]
Is the given example not enough?

**Comment by arthurfiorette** [comment]
No, I do not know what is `O.Either` neither have time to learn how to use `ts-toolbelt` just to solve this issue.

If you provide something I can clone, install and reproduce via a npm script would be help.

**Comment by eranelbaz** [comment]
I'll try to work on one soon
This is an old issue (almost 4 years) I also need to remember what is happening here
